prevent using files directly as path inputs

Otherwise it creates issues downstream:

```
imgbom ᓆ go run main.go -vvv dir:///Users/alfredo/.vimrc
[DEBUG]	Application config:
configpath: ""
presenteropt: 2
output: text
scopeopt: 2
scope: AllLayers
quiet: false
log:
  structured: false
  levelopt: debug
  level: ""
  filelocation: ""
clioptions:
  configpath: ""
  verbosity: 3

[DEBUG]	protocol: {Type:dir Value:/Users/alfredo/.vimrc}
[INFO]	Identifying Distro
[ERROR]	path (/Users/alfredo/.vimrc/etc/os-release) is not valid: stat /Users/alfredo/.vimrc/etc/os-release: not a directory
[ERROR]	path (/Users/alfredo/.vimrc/Users/alfredo/.vimrc/etc/os-release) is not valid: stat /Users/alfredo/.vimrc/Users/alfredo/.vimrc/etc/os-release: not a directory
[INFO]	no content present for ref: {%!s(file.ID=1) /Users/alfredo/.vimrc/etc/os-release}
[ERROR]	path (/Users/alfredo/.vimrc/usr/lib/os-release) is not valid: stat /Users/alfredo/.vimrc/usr/lib/os-release: not a directory
[ERROR]	path (/Users/alfredo/.vimrc/Users/alfredo/.vimrc/usr/lib/os-release) is not valid: stat /Users/alfredo/.vimrc/Users/alfredo/.vimrc/usr/lib/os-release: not a directory
[INFO]	no content present for ref: {%!s(file.ID=2) /Users/alfredo/.vimrc/usr/lib/os-release}
[ERROR]	path (/Users/alfredo/.vimrc/bin/busybox) is not valid: stat /Users/alfredo/.vimrc/bin/busybox: not a directory
[ERROR]	path (/Users/alfredo/.vimrc/Users/alfredo/.vimrc/bin/busybox) is not valid: stat /Users/alfredo/.vimrc/Users/alfredo/.vimrc/bin/busybox: not a directory
[INFO]	no content present for ref: {%!s(file.ID=3) /Users/alfredo/.vimrc/bin/busybox}
[ERROR]	error identifying distro
[INFO]	Creating the Catalog
[ERROR]	path (/Users/alfredo/.vimrc/var/lib/dpkg/status) is not valid: stat /Users/alfredo/.vimrc/var/lib/dpkg/status: not a directory
[DEBUG]	cataloger 'dpkg-cataloger' selected '1' files
[DEBUG]	cataloger 'bundler-cataloger' selected '1' files
[DEBUG]	cataloger 'python-cataloger' selected '1' files
[ERROR]	path (/Users/alfredo/.vimrc/var/lib/rpm/Packages) is not valid: stat /Users/alfredo/.vimrc/var/lib/rpm/Packages: not a directory
[DEBUG]	cataloger 'rpmdb-cataloger' selected '2' files
[DEBUG]	cataloger 'java-cataloger' selected '2' files
[ERROR]	path (/Users/alfredo/.vimrc/Users/alfredo/.vimrc/var/lib/dpkg/status) is not valid: stat /Users/alfredo/.vimrc/Users/alfredo/.vimrc/var/lib/dpkg/status: not a directory
[ERROR]	could not produce catalog: %!w(*errors.errorString=&{could not read contents of file: /Users/alfredo/.vimrc/var/lib/dpkg/status})
[Path: /Users/alfredo/.vimrc]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x411051b]

goroutine 68 [running]:
github.com/anchore/imgbom/imgbom/pkg.(*Catalog).Enumerate.func1(0xc01369a480, 0x0, 0x0, 0x0, 0x0)
	/Users/alfredo/python/imgbom/imgbom/pkg/catalog.go:76 +0x6b
created by github.com/anchore/imgbom/imgbom/pkg.(*Catalog).Enumerate
	/Users/alfredo/python/imgbom/imgbom/pkg/catalog.go:74 +0x84
exit status 2
```

--- a/syft/scope/scope.go
+++ b/syft/scope/scope.go
@@ -2,6 +2,7 @@ package scope
 
 import (
 	"fmt"
+	"os"
 
 	"github.com/anchore/stereoscope"
 
@@ -32,7 +33,11 @@ func NewScope(userInput string, o Option) (Scope, func(), error) {
 
 	switch protocol.Type {
 	case directoryProtocol:
-		// populate the scope object for dir
+		err := isValidPath(protocol.Value)
+		if err != nil {
+			return Scope{}, func() {}, fmt.Errorf("unable to process path, must exist and be a directory: %w", err)
+		}
+
 		s, err := NewScopeFromDir(protocol.Value, o)
 		if err != nil {
 			return Scope{}, func() {}, fmt.Errorf("could not populate scope from path (%s): %w", protocol.Value, err)
@@ -114,4 +119,19 @@ func (s Scope) Source() interface{} {
 	}
 
 	return nil
+}
+
+// isValidPath ensures that the user-provided input will correspond to a path
+// that exists and is a directory
+func isValidPath(userInput string) error {
+	fileMeta, err := os.Stat(userInput)
+	if err != nil {
+		return err
+	}
+
+	if fileMeta.IsDir() {
+		return nil
+	}
+
+	return fmt.Errorf("path is not a directory: %s", userInput)
 }

--- a/syft/scope/scope_test.go
+++ b/syft/scope/scope_test.go
@@ -213,3 +213,40 @@ func TestFilesByGlob(t *testing.T) {
 		})
 	}
 }
+
+func TestIsValidPath(t *testing.T) {
+	testCases := []struct {
+		desc    string
+		input   string
+		isError bool
+	}{
+		{
+			desc:    "path is valid",
+			input:   "test-fixtures",
+			isError: false,
+		},
+		{
+			desc:    "file is invalid",
+			input:   "test-fixtures/.vimrc",
+			isError: true,
+		},
+		{
+			desc:    "path does not exist",
+			input:   "foo/bar/baz",
+			isError: true,
+		},
+	}
+	for _, test := range testCases {
+		t.Run(test.desc, func(t *testing.T) {
+			err := isValidPath(test.input)
+			if err != nil && !test.isError {
+				t.Errorf("did not expect and error, got: %w", err)
+			}
+
+			if err == nil && test.isError {
+				t.Errorf("expected an error but didn't get one")
+			}
+
+		})
+	}
+}

